apollo-ast/src/main/kotlin/com/apollographql/apollo3/ast/check_key_fields.kt

```diff
@@ -53,6 +67,15 @@ private fun CheckKeyFieldsScope.checkFieldSet(path: String, selections: List<GQL
 
 private class FieldWithParent(val field: GQLField, val parentType: String)
 
+private val collectFieldsCache = mutableMapOf<String, List<FieldWithParent>>()
+private fun CheckKeyFieldsScope.collectFieldsCached(
+    selections: List<GQLSelection>,
+    parentType: String,
+    implementedTypes: Set<String>,
+) = collectFieldsCache.getOrPut("$selections $parentType $implementedTypes") {
+  collectFields(selections, parentType, implementedTypes)
```

Comment:
Do you have any idea the cache hit ratio for this? I'd expect it to be relatively low?

Reply:
Withe the test in the PR I can see that the method was run `2693` times overall, and only `405` times after making it cached. So `2288` duplicated calls, meaning ~80% cache hit?